rocAL/rocAL_pybind/example/coco_pipeline.py

```diff
@@ -160,131 +171,75 @@ def __next__(self):
+#Image id of a batch of images
+        self.image_id = np.zeros(self.bs, dtype="int32")
+        self.loader.GetImageId(self.image_id)
 # Count of labels/ bboxes in a batch
         self.bboxes_label_count = np.zeros(self.bs, dtype="int32")
         self.count_batch = self.loader.GetBoundingBoxCount(
             self.bboxes_label_count)
         print("Count Batch:", self.count_batch)
-# 1D labels array in a batch
-        self.labels = np.zeros(self.count_batch, dtype="int32")
-        self.loader.GetBBLabels(self.labels)
-        print(self.labels)
-# 1D bboxes array in a batch
-        self.bboxes = np.zeros((self.count_batch*4), dtype="float32")
-        self.loader.GetBBCords(self.bboxes)
-        print(self.bboxes)
+# 1D labels & bboxes array
+        self.encoded_bboxes = np.zeros((self.count_batch*4), dtype="float32")
+        self.encoded_labels = np.zeros(self.count_batch, dtype="int32")
+        self.loader.copyEncodedBoxesAndLables(
+            self.encoded_bboxes, self.encoded_labels)
 # Image sizes of a batch
         self.img_size = np.zeros((self.bs * 2), dtype="int32")
         self.loader.GetImgSizes(self.img_size)
         print("Image sizes:", self.img_size)
-        count = 0
-        sum_count = 0
+
+        encoded_bboxes_tensor = torch.tensor(
+            self.encoded_bboxes).view(self.bs, -1, 4).contiguous()
+        encodded_labels_tensor = torch.tensor(
+            self.encoded_labels).long().view(self.bs, -1)
+        image_id_tensor = torch.tensor(self.image_id)
+        image_size_tensor = torch.tensor(self.img_size).view(-1, self.bs, 2)
         for i in range(self.bs):
-            count = self.bboxes_label_count[i]
-            print("labels:", self.labels[sum_count: sum_count+count])
-            print("bboxes:", self.bboxes[sum_count*4: (sum_count+count)*4])
-            print("Image w & h:", self.img_size[i*2:(i*2)+2])
-            print("Image names:", self.Img_name[i*16:(i*16)+12])
-            self.img_name = self.Img_name[i*16:(i*16)+12]
-            self.img_name = self.img_name.decode('utf_8')
-            self.img_name = np.char.lstrip(self.img_name, chars='0')
-            print("Image names:", self.img_name)
-            self.label_2d_numpy = (self.labels[sum_count: sum_count+count])
-            if(self.loader._BoxEncoder != True):
-                self.label_2d_numpy = np.reshape(
-                    self.label_2d_numpy, (-1, 1)).tolist()
-            self.bb_2d_numpy = (self.bboxes[sum_count*4: (sum_count+count)*4])
-            self.bb_2d_numpy = np.reshape(self.bb_2d_numpy, (-1, 4)).tolist()
-            # Draw images: make sure to revert the mean and std to 0 and 1 for displaying original images without normalization
+            index_list = []
+            actual_bboxes = []
+            actual_labels = []
+            for idx, x in enumerate(encodded_labels_tensor[i]):
+                if x != 0:
+                    index_list.append(idx)
+                    actual_bboxes.append(
```

Comment:
please don't split statements into multiple lines unless it is too long.. I see a lot in this file

Reply:
Okay